fix: preserve existing blob hashes when 'X-Goog-Hash header' is not present

MartinKilonzo opened the pull request:
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #266 🦕

tseaver commented on google/cloud/storage/blob.py:
```diff
@@ -818,8 +818,8 @@ def _extract_headers_from_download(self, response):
-        self.crc32c = digests.get("crc32c", None)
-        self.md5_hash = digests.get("md5", None)
+        self.crc32c = digests.get("crc32c", self.crc32c)
+        self.md5_hash = digests.get("md5", self.md5_hash)
```
Likely pedantic, but I would prefer to have the assignment to the local properties conditioned by the presence of the `X-Goog-Hash` header:  if it is present, but does not contain one of the elements, that element should be cleared as well.  So, I'm arguing for having lines 814-822 contained inside an `if x_goog_hash:` block.

MartinKilonzo replied:
I like it. I think that better expresses the intent.

Change made in [this commit](https://github.com/googleapis/python-storage/pull/267/commits/f0e77dcab1caca2677b14cc91e72ffa266f8ad0a).

tseaver commented on tests/unit/test_blob.py:
```diff
@@ -1542,10 +1542,16 @@ def test_download_as_string_w_response_headers(self):
 
     def test_download_as_string_w_hash_response_header_none(self):
```
We should be renaming this method to correspond with its new semantics, e.g.:

```suggestion
    def test_download_as_string_w_hash_response_header_missing(self):
```

MartinKilonzo commented:
Thanks for the review!
Added the proposed changes. @tseaver, would you mind checking your CLA per @googlebot: 

> We found a Contributor License Agreement for you (the sender of this pull request), but were unable to find agreements for all the commit author(s) or Co-authors. If you authored these, maybe you used a different email address in the git commits than was used to sign the CLA ([login here](https://cla.developers.google.com/) to double check)? If these were authored by someone else, then they will need to sign a CLA as well, and confirm that they're okay with these being contributed to Google.
> In order to pass this check, please resolve this problem and then comment `@googlebot I fixed it.`. If the bot doesn't comment, it means it doesn't think anything has changed.
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fgoogleapis%2Fpython-storage%2Fpull%2F267) for more info**.

tseaver commented:
@googlebot I consent

tseaver commented:
@googlebot I consent

tseaver commented:
@MartinKilonzo Thanks for persevering!
